tests/lineardev_tests.rs

```diff
@@ -0,0 +1,177 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#[macro_use]
+extern crate log;
+extern crate uuid;
+extern crate devicemapper;
+extern crate libstratis;
+#[macro_use]
+mod util;
+
+use devicemapper::DM;
+
+use libstratis::engine::strat_engine::blockdev;
+use libstratis::engine::strat_engine::blockdev::BlockDev;
+use libstratis::engine::strat_engine::lineardev::LinearDev;
+use libstratis::engine::strat_engine::metadata::MIN_MDA_SECTORS;
+use libstratis::types::Sectors;
+
+use std::iter::FromIterator;
+use std::path::Path;
+use std::thread;
+use std::time::Duration;
+
+use util::blockdev_utils::clean_blockdev_headers;
+use util::blockdev_utils::get_size;
+use util::test_config::TestConfig;
+use util::test_consts::DEFAULT_CONFIG_FILE;
+use util::test_result::TestError::Framework;
+use util::test_result::TestErrorEnum::Error;
+use util::test_result::TestResult;
+
+use uuid::Uuid;
+
+fn concat_blockdevs(dm: &DM, name: &str, block_devs: &Vec<&BlockDev>) -> TestResult<LinearDev> {
+
+    let mut linear_dev = LinearDev::new_lineardev(name);
+
+    match linear_dev.concat(dm, block_devs) {
+        Ok(_) => return Ok(linear_dev),
+        Err(e) => {
+            // tear down a partially created DM device if it exists
+            match linear_dev.teardown(&dm) {
+                Ok(_) => info!("completed teardown of {}", name),
+                Err(_) => info!("failed teardown of {}", name),
+            }
+            let message = format!("linear_dev.concat failed : {:?}", e);
+            return Err(Framework(Error(message)));
+        }
+
+    }
+}
+
+fn validate_sizes(name: &str, block_devs: &Vec<&BlockDev>) -> TestResult<()> {
+
+    let mut linear_sectors = Sectors(0);
+
+    for blockdev in block_devs {
+        let (_start_sector, length) = blockdev.avail_range();
+        linear_sectors = linear_sectors + length;
+    }
+
+    debug!("available linear space = {} sectors", linear_sectors);
+
+    // The /dev/mapper/<name> device is not immediately available for use.
```

Comment:
Yeah, I suspect there is... we could probably wait for an event, poll etc.

@agrover suggested this was good enough for a test.  I'm happy to put a TODO to remind us to fix it.  That work?